code/modules/assembly/flash.dm

```diff
@@ -90,6 +90,8 @@
 	return TRUE
 
 /obj/item/device/assembly/flash/proc/flash_carbon(mob/living/carbon/M, mob/user, power = 15, targeted = TRUE, generic_message = FALSE)
+	if(!M || !iscarbon(M))

```

Comment:
`!M` is already implied in `!iscarbon`, and just an `!istype(M)` would do here since var is already typecasted